src/icp_prototype_backend/src/lib.rs

```diff
@@ -487,13 +545,22 @@ async fn call_query_blocks() {
                         Err(_) => "HASH-IS-NOT-AVAILABLE".to_string(),
                     };
                     ic_cdk::println!("Hash: {:?}", hash);
-                    let transaction =
-                        StoredTransactions::new(block_count, block.transaction.clone(), hash);
+                    let transaction = StoredTransactions::new(
+                        block_count,
+                        block.transaction.clone(),
+                        hash.clone(),
+                    );
 
                     if !transactions.contains_key(&block_count) {
                         // Filter keys that exist
                         ic_cdk::println!("Inserting transaction");
                         let _ = transactions.insert(block_count, transaction);
+
+                        // Track the first block hash in the iter
+                        if block_count == next_block {
```

Comment:
I assumed we only populate `first_block_hash` with value once, which then we could simply check `first_block_hash.is_empty()` to assign the value once, wdyt?

Reply:
SMART!